pkg/pdatatest/pmetrictest/options.go

```diff
@@ -159,6 +159,64 @@ func roundDataPointSliceValues(dataPoints pmetric.NumberDataPointSlice, precisio
 	}
 }
 
+// IgnoreExemplars is a CompareMetricsOption that clears exemplar fields on all the Sum data points.
+func IgnoreExemplars() CompareMetricsOption {
+	return compareMetricsOptionFunc(func(expected, actual pmetric.Metrics) {
+		maskExemplars(expected)
+		maskExemplars(actual)
+	})
+}
+
+func maskExemplars(metrics pmetric.Metrics) {
+	rms := metrics.ResourceMetrics()
+	for i := 0; i < rms.Len(); i++ {
+		ilms := rms.At(i).ScopeMetrics()
+		for j := 0; j < ilms.Len(); j++ {
+			metrics := ilms.At(j).Metrics()
+			for g := 0; g < metrics.Len(); g++ {
+				switch metrics.At(g).Type() {
+				case pmetric.MetricTypeGauge:
+					datapoints := metrics.At(g).Gauge().DataPoints()
+					for k := 0; k < datapoints.Len(); k++ {
+						exemplars := datapoints.At(k).Exemplars()
+						for f := 0; f < exemplars.Len(); f++ {
+							emptyExemplar := pmetric.NewExemplar()
+							emptyExemplar.CopyTo(exemplars.At(f))
+						}
+					}
+				case pmetric.MetricTypeSum:
+					datapoints := metrics.At(g).Sum().DataPoints()
+					for k := 0; k < datapoints.Len(); k++ {
+						exemplars := datapoints.At(k).Exemplars()
+						for f := 0; f < exemplars.Len(); f++ {
```

Comment:
Should use the new .All() iterators for all ranging, rather than the older style for i < Len() + .At(i).